### DISCUSSION TITLE
url

### DISCUSSION THREAD

**Discussion by qwertyuiop5t** [norkunas/youtube-dl-php#193] [Q&A]
can you modify the download code such that it only generate url so that i can stream it in my web browser

**Comment by norkunas** [comment]
There is no need to modify anything, just use `skip-download` option

**Reply by qwertyuiop5t** [reply]
I am learning php and don't no how to use skip-downlaod and get only url that's why i am telling you to write me the code which gives just stream link

**Reply by norkunas** [reply]
Why are you telling me to write code? Am I working for you?

There is an `Options` object which you need to configure, and if you are using proper IDE you would get autocomplete hints when trying to write `Options::create()->skip..` so the complete usage is `Options::create()->skipDownload(true)`.

Also please don't create duplicate issues. And also this is just your question about usage and not the issue with the library itself.

**Comment by qwertyuiop5t** [comment]
you are not working me. I have just ask for education purpose. Sorry if you
get it wrong. Does not know how to use options that why i am asking you.
Sorry!!

**Reply by norkunas** [reply]
Just try to ask in a friendlier tone and I will try to help :)

**Comment by qwertyuiop5t** [comment]
Does not know much about php. Do you have any tutorial about
youtube-dl-php. Please. If you have any youtube channel. Please sir

**Reply by norkunas** [reply]
Most of the available options are in readme https://github.com/yt-dlp/yt-dlp and they are named in this lib in similar

**Comment by qwertyuiop5t** [comment]
can you please help me out. i want to get video url. done everything. but
the code did not return me any thing,. it shows a blank page.
<?php

declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

use YoutubeDl\Options;
use YoutubeDl\YoutubeDl;

$yt = new YoutubeDl();
$collection = $yt->download(
   Options::create()->skipDownload(true)
    ->downloadPath('/path/to/downloads')
    ->url('https://www.youtube.com/watch?v=oDAw7vW7H0c')
);

foreach ($collection->getVideos() as $video) {
    if ($video->getError() !== null) {
        echo "Error downloading video: {$video->getError()}.";
    } else {
        $video->getUrl(); // audio file
    }
}
?>

**Reply by norkunas** [reply]
Is your `youtube-dl` or `yt-dlp` library available globally? Otherwise you have to set it `$yt->setBinPath('/path/to/youtube-dl');`
